lib/net/http.rb

```diff
@@ -1024,7 +1028,13 @@ def connect
       debug "opened"
       if use_ssl?
         if proxy?
-          plain_sock = BufferedIO.new(s, read_timeout: @read_timeout,
+          if @proxy_use_ssl
+            proxy_sock = OpenSSL::SSL::SSLSocket.new(s)
```

Comment:
this probably needs some more work, especially when it comes to verification etc, but it's a start :)

Reply:
@evgeni, I try to avoid code changes.

Reply:
Not sure what you mean by that?